sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallingServerClientBuilder.java

```diff
@@ -0,0 +1,346 @@
+            String endpoint = connectionStringObject.getEndpoint();
+            String accessKey = connectionStringObject.getAccessKey();
+            endpoint(endpoint).credential(new AzureKeyCredential(accessKey));
+        }
+
+        Objects.requireNonNull(endpoint);
+        if (pipeline == null) {
+            Objects.requireNonNull(httpClient);
+        }
+
+        HttpPipeline builderPipeline = pipeline;
+        if (pipeline == null) {
+            builderPipeline = createHttpPipeline(httpClient);
+        }
+
+        AzureCommunicationCallingServerServiceImplBuilder clientBuilder = new AzureCommunicationCallingServerServiceImplBuilder();
+        clientBuilder.endpoint(endpoint).pipeline(builderPipeline);
+
+        return clientBuilder.buildClient();
+    }
+
+    /**
+     * Allows the user to set a variety of client-related options, such as
+     * user-agent string, headers, etc.
+     *
+     * @param clientOptions object to be applied
+     * @return The updated {@link CallingServerClientBuilder} object.
+     */
+    public CallingServerClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (tokenCredential != null && azureKeyCredential != null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Both 'credential' and 'keyCredential' are set. Just one may be used."));
+        }
+        if (tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(tokenCredential,
+                "https://communication.azure.com//.default");
+        } else if (azureKeyCredential != null) {
+            return new HmacAuthenticationPolicy(azureKeyCredential);
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
+        }
+    }
+
+    private HttpPipeline createHttpPipeline(HttpClient httpClient) {
+        if (pipeline != null) {
+            return pipeline;
+        }
+
+        List<HttpPipelinePolicy> policyList = new ArrayList<>();
+
+        ClientOptions buildClientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
+        HttpLogOptions buildLogOptions = (httpLogOptions == null) ? new HttpLogOptions() : httpLogOptions;
+
+        String applicationId = null;
+        if (!CoreUtils.isNullOrEmpty(buildClientOptions.getApplicationId())) {
+            applicationId = buildClientOptions.getApplicationId();
+        } else if (!CoreUtils.isNullOrEmpty(buildLogOptions.getApplicationId())) {
+            applicationId = buildLogOptions.getApplicationId();
+        }
+
+        // Add required policies
+        policyList.add(createHttpPipelineAuthPolicy());
```

Comment:
Please move auth policy to after retry policy. We made a mistake in other SDKs wrt this.

Reply:
Auth policy is moved after retry policy